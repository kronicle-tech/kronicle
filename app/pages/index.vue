<template>
  <div class="m-3">
    <Intro id="intro" :title="introTitle" :markdown="introMarkdown" />

    <b-card-group columns>
      <b-card
        header="Kronicle"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Kronicle contains the following:
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-areas' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Areas</b>
            <span class="lead">
              <b-badge class="area-count" variant="primary" pill>{{
                  areaCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-teams' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Teams</b>
            <span class="lead">
              <b-badge class="team-count" variant="primary" pill>{{
                  teamCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Components</b>
            <span class="lead">
              <b-badge class="component-count" variant="primary" pill>{{
                  componentCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tech-debts' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Tech Debts</b>
            <span class="lead">
              <b-badge class="tech-debt-count" variant="primary" pill>{{
                  techDebtCount
                }}</b-badge>
            </span>
          </b-list-group-item>

          <b-list-group-item
            :to="{ name: 'all-components-tests' }"
            class="d-flex justify-content-between align-items-center"
          >
            <b>Test Results</b>
            <span class="lead">
              <b-badge class="test-result-count" variant="primary" pill>{{
                  testResultCount
                }}</b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Scanners"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Scanners dynamically add data to Kronicle at runtime:
          </b-list-group-item>

          <b-list-group-item v-for="scanner in scanners" :key="scanner.id">
            <b class="scanner-id text-info">{{ scanner.id }}</b>
            <br />
            <Markdown :markdown="scanner.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tests"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            Tests evaluate the data associated with components in Kronicle and
            return outcomes of pass, fail and not applicable
          </b-list-group-item>

          <b-list-group-item
            v-for="test in tests"
            :key="test.id"
            :to="{
              name: 'tests-testId',
              params: { testId: test.id },
            }"
          >
            <b class="test-id text-info">{{ test.id }}</b>
            <br />
            <Markdown :markdown="test.description" class="description" />
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Platforms"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group-item
          class="d-flex justify-content-between align-items-center text-muted"
        >
          The following platforms are associated with components in Kronicle:
        </b-list-group-item>

        <b-list-group flush>
          <b-list-group-item
            v-for="platformCount in platformCounts"
            :key="platformCount.item"
            :to="{
              name: 'all-components',
              query: { platformId: platformCount.item || 'undefined' },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="platform">{{ platformCount.item || 'missing' }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ platformCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Tags"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            The components in Kronicle are tagged with the following:
          </b-list-group-item>

          <b-list-group-item
            v-for="tagCount in tagCounts"
            :key="tagCount.item"
            :to="{
              name: 'all-components',
              query: { tag: tagCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="tag">{{ tagCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ tagCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>

      <b-card
        header="Component Types"
        header-bg-variant="info"
        header-class="lead"
        no-body
      >
        <b-list-group flush>
          <b-list-group-item
            class="d-flex justify-content-between align-items-center text-muted"
          >
            There are the following types of components in Kronicle:
          </b-list-group-item>

          <b-list-group-item
            v-for="componentTypeCount in componentTypeCounts"
            :key="componentTypeCount.item"
            :to="{
              name: 'all-components',
              query: { componentType: componentTypeCount.item },
            }"
            class="d-flex justify-content-between align-items-center"
          >
            <b class="component-type">{{ componentTypeCount.item }}</b>
            <span class="lead">
              <b-badge variant="primary" pill>
                {{ componentTypeCount.count }}
              </b-badge>
            </span>
          </b-list-group-item>
        </b-list-group>
      </b-card>
    </b-card-group>
  </div>
</template>

<style scoped>
.description {
  display: inline-block;
  color: white;
  text-decoration: none;
}

.description >>> p {
  margin-bottom: 0;
}
</style>

<script lang="ts">
import Vue from 'vue'
import {
  BBadge,
  BCard,
  BCardGroup,
  BListGroup,
  BListGroupItem,
} from 'bootstrap-vue'
import { MetaInfo } from 'vue-meta'
import { Area, Component, Scanner, Team, Test } from '~/types/kronicle-service'
import { ItemCount, itemCounts } from '~/src/arrayUtils'
import Intro from '~/components/Intro.vue'
import Markdown from '~/components/Markdown.vue'

export default Vue.extend({
  components: {
    'b-badge': BBadge,
    'b-card': BCard,
    'b-card-group': BCardGroup,
    'b-list-group': BListGroup,
    'b-list-group-item': BListGroupItem,
    Intro,
    Markdown,
  },
  async asyncData({ $config }) {
    const areas = await fetch(
      `${$config.serviceBaseUrl}/v1/areas?fields=areas(id)`
    )
      .then((res) => res.json())
      .then((json) => json.areas as Area[])

    const teams = await fetch(
      `${$config.serviceBaseUrl}/v1/teams?fields=teams(id)`
    )
      .then((res) => res.json())
      .then((json) => json.teams as Team[])

    const components = await fetch(
      `${$config.serviceBaseUrl}/v1/components?fields=components(id,typeId,platformId,tags,techDebts(doesNotExist),testResults(doesNotExist))`
    )
      .then((res) => res.json())
      .then((json) => json.components as Component[])

    const scanners = await fetch(
      `${$config.serviceBaseUrl}/v1/scanners?fields=scanners(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.scanners as Scanner[])

    const tests = await fetch(
      `${$config.serviceBaseUrl}/v1/tests?fields=tests(id,description)`
    )
      .then((res) => res.json())
      .then((json) => json.tests as Test[])

    return {
      areas,
      teams,
      components,
      scanners,
      tests,
    }
  },
  data() {
    return {
      introTitle: this.$config.introTitle as String | undefined,
      introMarkdown: this.$config.introMarkdown as String | undefined,
      areas: [] as Area[],
      teams: [] as Team[],
      components: [] as Component[],
      scanners: [] as Scanner[],
      tests: [] as Test[],
    }
  },
  head(): MetaInfo {
    return {
      title: 'Kronicle',
    }
  },
  computed: {
    areaCount(): number {
      return this.areas.length
    },
    teamCount(): number {
      return this.teams.length
    },
    componentCount(): number {
      return this.components.length
    },
    componentTypeCounts(): ItemCount<string>[] {
      return itemCounts(this.components.map((component) => component.typeId))
    },
    platformCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.map((component) => component.platformId)
      )
    },
    tagCounts(): ItemCount<string>[] {
      return itemCounts(
        this.components.flatMap((component) => component.tags ?? [])
      )
    },
    techDebtCount(): number {
      return this.components.flatMap((component) => component.techDebts ?? [])
        .length
    },
    testResultCount(): number {
      return this.components.flatMap((component) => component.testResults ?? [])
        .length
    },
  },
})
</script>
