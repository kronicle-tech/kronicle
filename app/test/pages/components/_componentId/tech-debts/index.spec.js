import Index from '@/pages/components/_componentId/tech-debts/index.vue'
import { createPageWrapper } from '~/test/pages/pageUtils'
import {
  createComponent,
  createComponentAvailableDataRequests,
  createComponentWithTechDebts,
} from '~/test/testDataUtils'

describe('Index', () => {
  const route = {
    params: {
      componentId: 'test-component-id-1',
    },
  }
  let component
  let wrapper
  async function createWrapper() {
    wrapper = await createPageWrapper(Index, {
      route,
      serviceRequests: {
        ...createComponentAvailableDataRequests(),
        '/v1/components/test-component-id-1?fields=component(id,name,type,tags,teams,platformId,techDebts)':
          {
            responseBody: { component },
          },
      },
    })
  }

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when Get Component service endpoint returns a component', () => {
    beforeEach(() => {
      component = createComponent({ componentNumber: 1 })
    })

    test('has the right page title', async () => {
      await createWrapper()
      expect(wrapper.vm.$metaInfo.title).toBe(
        'Kronicle - Test Component Name 1 - Tech Debts'
      )
    })

    describe('when the component has no tech debts', () => {
      test('renders no tech debts', async () => {
        await createWrapper()
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('when the component has tech debts', () => {
      beforeEach(() => {
        component = createComponentWithTechDebts({ componentNumber: 1 })
      })

      test('renders the details of the tech debts', async () => {
        await createWrapper()
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })
})
