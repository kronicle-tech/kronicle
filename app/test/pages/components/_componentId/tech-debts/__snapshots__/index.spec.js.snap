// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index when Get Component service endpoint returns a component when the component has no tech debts renders no tech debts 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info my-3"
  >
    Test Component Name 1 - Tech Debts
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
       Info 
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/diagrams"
      target="_self"
    >
      
    Diagrams
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      
    Errors
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      
    Tests
  
    </a>
  </div>
   
  <div>
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <div
          class="list-group-item list-group-item-success"
        >
          <span
            class="display-1"
          >
            0
          </span>
           tech debts
      
        </div>
         
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-3 mb-4"
      components="[object Object]"
    >
       
      <button
        aria-controls="filters"
        aria-expanded="false"
        class="btn btn-primary collapsed"
        id="toggleFilters"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="blank"
          class="bi-blank b-icon bi"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
        </svg>
         Filters
  
      </button>
       
      <transition-stub
        class="mt-3"
        css="true"
        enteractiveclass="collapsing"
        enterclass=""
        entertoclass="collapse show"
        leaveactiveclass="collapsing"
        leaveclass="collapse show"
        leavetoclass="collapse"
      >
        <div
          class="collapse"
          id="filters"
          style="display: none;"
        >
          <div
            class="card-columns"
          >
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__51"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__51__BV_label_"
                    tabindex="-1"
                  >
                    Teams
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__52"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__52_BV_option_0"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__52_BV_option_0"
                        >
                          <span>
                            test-team-id-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__52_BV_option_1"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__52_BV_option_1"
                        >
                          <span>
                            test-team-id-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__55"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__55__BV_label_"
                    tabindex="-1"
                  >
                    Tags
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__56"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__56_BV_option_0"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__56_BV_option_0"
                        >
                          <span>
                            test-tag-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__56_BV_option_1"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__56_BV_option_1"
                        >
                          <span>
                            test-tag-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__59"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__59__BV_label_"
                    tabindex="-1"
                  >
                    Component Types
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__60"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__60_BV_option_0"
                          name="componentTypeId"
                          type="checkbox"
                          value="test-component-type-id-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__60_BV_option_0"
                        >
                          <span>
                            test-component-type-id-1
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__62"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__62__BV_label_"
                    tabindex="-1"
                  >
                    Platforms
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__63"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__63_BV_option_0"
                          name="platformId"
                          type="checkbox"
                          value="undefined"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__63_BV_option_0"
                        >
                          <span>
                            undefined
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <div
                  class="form-group"
                  id="__BVID__65"
                  role="group"
                >
                  <label
                    class="d-block"
                    for="component-filter"
                    id="__BVID__65__BV_label_"
                  >
                    Component
                  </label>
                  <div>
                    <select
                      class="custom-select custom-select-sm"
                      id="component-filter"
                    >
                      <option
                        value=""
                      >
                        Please select a component
                      </option>
                      <option
                        value="test-component-id-1"
                      >
                        test-component-id-1
                      </option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
     
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          0 Tech Debts
        </h4>
        <!---->
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;

exports[`Index when Get Component service endpoint returns a component when the component has tech debts renders the details of the tech debts 1`] = `
<div
  class="m-3"
>
  <h1
    class="text-info my-3"
  >
    Test Component Name 1 - Tech Debts
  </h1>
   
  <div
    class="my-3"
  >
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1"
      target="_self"
    >
       Info 
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/diagrams"
      target="_self"
    >
      
    Diagrams
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/cross-functional-requirements"
      target="_self"
    >
      
    Cross Functional Requirements
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tech-debts"
      target="_self"
    >
      
    Tech Debts
  
    </a>
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/errors"
      target="_self"
    >
      
    Errors
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/lines-of-code"
      target="_self"
    >
      
    Lines of Code
  
    </a>
     
    <!---->
     
    <!---->
     
    <!---->
     
    <a
      class="btn my-1 btn-secondary"
      href="/components/test-component-id-1/tests"
      target="_self"
    >
      
    Tests
  
    </a>
  </div>
   
  <div>
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="list-group"
      >
        <!---->
         
        <div
          class="list-group-item list-group-item-warning"
        >
          <span
            class="display-2"
          >
            1
          </span>
           
          <b>
            High
          </b>
           tech debt
      
        </div>
        <div
          class="list-group-item list-group-item-info"
        >
          <span
            class="display-3"
          >
            1
          </span>
           
          <b>
            Medium
          </b>
           tech debt
      
        </div>
      </div>
      <!---->
      <!---->
    </div>
     
    <div
      class="mt-3 mb-4"
      components="[object Object]"
    >
       
      <button
        aria-controls="filters"
        aria-expanded="false"
        class="btn btn-primary collapsed"
        id="toggleFilters"
        type="button"
      >
        <svg
          aria-hidden="true"
          aria-label="blank"
          class="bi-blank b-icon bi"
          fill="currentColor"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g />
        </svg>
         Filters
  
      </button>
       
      <transition-stub
        class="mt-3"
        css="true"
        enteractiveclass="collapsing"
        enterclass=""
        entertoclass="collapse show"
        leaveactiveclass="collapsing"
        leaveclass="collapse show"
        leavetoclass="collapse"
      >
        <div
          class="collapse"
          id="filters"
          style="display: none;"
        >
          <div
            class="card-columns"
          >
             
            <!---->
             
            <!---->
             
            <!---->
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__84"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__84__BV_label_"
                    tabindex="-1"
                  >
                    Teams
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__85"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__85_BV_option_0"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__85_BV_option_0"
                        >
                          <span>
                            test-team-id-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__85_BV_option_1"
                          name="team"
                          type="checkbox"
                          value="test-team-id-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__85_BV_option_1"
                        >
                          <span>
                            test-team-id-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__88"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__88__BV_label_"
                    tabindex="-1"
                  >
                    Tags
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__89"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__89_BV_option_0"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__89_BV_option_0"
                        >
                          <span>
                            test-tag-1-1
                          </span>
                        </label>
                      </div>
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__89_BV_option_1"
                          name="tag"
                          type="checkbox"
                          value="test-tag-1-2"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__89_BV_option_1"
                        >
                          <span>
                            test-tag-1-2
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__92"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__92__BV_label_"
                    tabindex="-1"
                  >
                    Component Types
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__93"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__93_BV_option_0"
                          name="componentTypeId"
                          type="checkbox"
                          value="test-component-type-id-1"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__93_BV_option_0"
                        >
                          <span>
                            test-component-type-id-1
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <fieldset
                  class="form-group"
                  id="__BVID__95"
                >
                  <legend
                    class="bv-no-focus-ring col-form-label pt-0"
                    id="__BVID__95__BV_label_"
                    tabindex="-1"
                  >
                    Platforms
                  </legend>
                  <div>
                    <div
                      class="bv-no-focus-ring"
                      id="__BVID__96"
                      role="group"
                      tabindex="-1"
                    >
                      <div
                        class="custom-control custom-checkbox"
                      >
                        <input
                          class="custom-control-input"
                          id="__BVID__96_BV_option_0"
                          name="platformId"
                          type="checkbox"
                          value="undefined"
                        />
                        <label
                          class="custom-control-label"
                          for="__BVID__96_BV_option_0"
                        >
                          <span>
                            undefined
                          </span>
                        </label>
                      </div>
                    </div>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </fieldset>
              </div>
              <!---->
              <!---->
            </div>
             
            <div
              class="card bg-secondary"
            >
              <!---->
              <!---->
              <div
                class="card-body p-2"
              >
                <!---->
                <!---->
                <div
                  class="form-group"
                  id="__BVID__98"
                  role="group"
                >
                  <label
                    class="d-block"
                    for="component-filter"
                    id="__BVID__98__BV_label_"
                  >
                    Component
                  </label>
                  <div>
                    <select
                      class="custom-select custom-select-sm"
                      id="component-filter"
                    >
                      <option
                        value=""
                      >
                        Please select a component
                      </option>
                      <option
                        value="test-component-id-1"
                      >
                        test-component-id-1
                      </option>
                    </select>
                    <!---->
                    <!---->
                    <!---->
                  </div>
                </div>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </transition-stub>
    </div>
     
    <div
      class="card my-3"
    >
      <!---->
      <!---->
      <div
        class="card-body"
      >
        <h4
          class="card-title"
        >
          2 Tech Debts
        </h4>
        <!---->
        <table
          class="table table-dark"
        >
          <thead>
            <tr>
              <th
                class="component"
              >
                Component
              </th>
               
              <th
                class="priority"
              >
                Priority
              </th>
               
              <th
                class="teams"
              >
                Teams
              </th>
               
              <th
                class="description"
              >
                Description
              </th>
               
              <th
                class="notes"
              >
                Notes
              </th>
               
              <th
                class="links"
              >
                Links
              </th>
            </tr>
          </thead>
           
          <tbody>
            <tr>
              <td
                class="component table-primary"
              >
                <a
                  href="/components/test-component-id-1"
                >
                  Test Component Name 1
                </a>
              </td>
               
              <td
                class="priority"
              >
                <span
                  class="badge badge-warning"
                >
                  
          High
        
                </span>
              </td>
               
              <td
                class="teams"
              >
                <ul>
                  <li>
                    <a
                      class="team-name"
                      href="/teams/test-team-id-1-1"
                    >
                      test-team-id-1-1
                    </a>
                     
                    <!---->
                     
                    <!---->
                  </li>
                  <li>
                    <a
                      class="team-name"
                      href="/teams/test-team-id-1-2"
                    >
                      test-team-id-1-2
                    </a>
                     
                    <!---->
                     
                    <!---->
                  </li>
                </ul>
              </td>
               
              <td
                class="description"
              >
                <div>
                  

                  <p>
                    Test Tech Debt Description 1 2
                  </p>
                </div>
              </td>
               
              <td
                class="notes"
              >
                <div>
                  

                  <p>
                    Test Tech Debt Notes 1 2
                  </p>
                </div>
              </td>
               
              <td
                class="links"
              >
                <ul>
                  <li>
                    <a
                      href="https://example.com/test-tech-debt-link-1-2-1"
                      target="_blank"
                    >
                      https://example.com/test-tech-debt-link-1-2-1
                    </a>
                     
                    <br />
                     
                    <div>
                      

                      <p>
                        Test Tech Debt Link Description 1 2 1
                      </p>
                    </div>
                  </li>
                  <li>
                    <a
                      href="https://example.com/test-tech-debt-link-1-2-2"
                      target="_blank"
                    >
                      https://example.com/test-tech-debt-link-1-2-2
                    </a>
                     
                    <br />
                     
                    <div>
                      

                      <p>
                        Test Tech Debt Link Description 1 2 2
                      </p>
                    </div>
                  </li>
                </ul>
              </td>
            </tr>
            <tr>
              <td
                class="component table-primary"
              >
                <a
                  href="/components/test-component-id-1"
                >
                  Test Component Name 1
                </a>
              </td>
               
              <td
                class="priority"
              >
                <span
                  class="badge badge-info"
                >
                  
          Medium
        
                </span>
              </td>
               
              <td
                class="teams"
              >
                <ul>
                  <li>
                    <a
                      class="team-name"
                      href="/teams/test-team-id-1-1"
                    >
                      test-team-id-1-1
                    </a>
                     
                    <!---->
                     
                    <!---->
                  </li>
                  <li>
                    <a
                      class="team-name"
                      href="/teams/test-team-id-1-2"
                    >
                      test-team-id-1-2
                    </a>
                     
                    <!---->
                     
                    <!---->
                  </li>
                </ul>
              </td>
               
              <td
                class="description"
              >
                <div>
                  

                  <p>
                    Test Tech Debt Description 1 1
                  </p>
                </div>
              </td>
               
              <td
                class="notes"
              >
                <div>
                  

                  <p>
                    Test Tech Debt Notes 1 1
                  </p>
                </div>
              </td>
               
              <td
                class="links"
              >
                <ul>
                  <li>
                    <a
                      href="https://example.com/test-tech-debt-link-1-1-1"
                      target="_blank"
                    >
                      https://example.com/test-tech-debt-link-1-1-1
                    </a>
                     
                    <br />
                     
                    <div>
                      

                      <p>
                        Test Tech Debt Link Description 1 1 1
                      </p>
                    </div>
                  </li>
                  <li>
                    <a
                      href="https://example.com/test-tech-debt-link-1-1-2"
                      target="_blank"
                    >
                      https://example.com/test-tech-debt-link-1-1-2
                    </a>
                     
                    <br />
                     
                    <div>
                      

                      <p>
                        Test Tech Debt Link Description 1 1 2
                      </p>
                    </div>
                  </li>
                </ul>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
